### PR TITLE
OCD-1428: Limit activity searches to a configurable max date range (currently set to 60 days)

### PR DESCRIPTION
**Pull request by dlucas1**

### CODE REVIEW

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/web/controller/ActivityController.java [hunk 1]
```diff
@@ -1009,4 +1037,15 @@
...
 	}
+	
+	private void validateActivityDates(Long startDate, Long endDate) throws ValidationException{
+		Calendar calendarCounter = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
+		Integer maxActivityRangeInDays = Integer.getInteger(env.getProperty("maxActivityRangeInDays"), 60);
+		calendarCounter.setTime(new Date(endDate));
+		calendarCounter.add(Calendar.DATE, -maxActivityRangeInDays);
+		if(new Date(startDate).compareTo(calendarCounter.getTime()) < 0){
+			throw new ValidationException("Cannot search an activity with a date range more than " + maxActivityRangeInDays + " days.");
+		}
```
Comment (andlar) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/web/controller/ActivityController.java]: What happens if startDate is _after_ endDate_?

File: chpl/chpl-service/src/main/resources/environment.properties.template [hunk 2]
```diff
@@ -9,6 +9,7 @@ questionableActivityThresholdDays=0
 summaryEmail=dlucas@ainq.com
 summaryEmailName=summaryStatistics.csv
 summaryEmailPeriodInDays=7
+maxActivityRangeInDays=60
```
Comment (andlar) [inline chpl/chpl-service/src/main/resources/environment.properties.template]: Stylistically, this properties file is getting hard to read. The section breaks with whitespace are useful

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/web/controller/ActivityController.java (hunk 1 above)
Comment (dlucas1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/web/controller/ActivityController.java]: Nothing is returned in the result.

Comment (andlar) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/web/controller/ActivityController.java]: A better user experience would be to have an error message show up

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/web/controller/ActivityController.java [hunk 3]
```diff
@@ -1042,6 +1042,10 @@ private void validateActivityDates(Long startDate, Long endDate) throws Validati
 		Calendar calendarCounter = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
 		Integer maxActivityRangeInDays = Integer.getInteger(env.getProperty("maxActivityRangeInDays"), 60);
 		calendarCounter.setTime(new Date(endDate));
+		if(new Date(startDate).compareTo(calendarCounter.getTime()) > 0){
+			throw new ValidationException("Cannot search an activity with a start date greater than the end date");
```
Comment (andlar) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/web/controller/ActivityController.java]: We should try to make errors that end users see as close to English as possible. I'd phrase this error as "Cannot search for activity with the start date after the end date"

**Pull request merged**